File: multi_cam_image_capture.py
from openpyxl import Workbook  
import time
import os 
from datetime import datetime
dir='/home/pi/indrone/test5/finalc1' 
wb = Workbook()  
sheet = wb.active 
n =0
sheet[f'A{n+1}'] = "Start time"
sheet[f'B{n+1}'] = "End time"
sheet[f'c{n+1}'] = "Total time required"
  

def function1():
          global n
          n += 1
          now = datetime.now().strftime('%H:%M:%S.%f')[:-3]
          
          sheet[f'A{n+1}'] = now
          # Execute i2cset command
          i2cset_cmd = 'i2cset -y 10 0x24 0x24 0x02'
          os.system(i2cset_cmd)

          # Execute libcamera-still command
          libcamera_cmd = f'libcamera-still --immediate --width 1920 --height 1080 -e png -o {dir}/f{n}.png --mode 3264:2448 --nopreview'
          os.system(libcamera_cmd)

          now = datetime.now().strftime('%H:%M:%S.%f')[:-3]
          sheet[f'B{n+1}'] = now

def function2():
          global n
          n += 1
          now = datetime.now().strftime('%H:%M:%S.%f')[:-3]
          sheet[f'A{n+1}'] = now
          i2cset_cmd = 'i2cset -y 10 0x24 0x24 0x12'
          os.system(i2cset_cmd)

          # Execute libcamera-still command
          libcamera_cmd = f'libcamera-still --immediate --width 1920 --height 1080 -e png  -o {dir}/ff{n}.png  --mode 3264:2448 --nopreview'
          os.system(libcamera_cmd)

          now = datetime.now().strftime('%H:%M:%S.%f')[:-3]
          sheet[f'B{n+1}'] = now


def function3():
          global n
          n += 1
          now = datetime.now().strftime('%H:%M:%S.%f')[:-3]
          sheet[f'A{n+1}'] = now
          i2cset_cmd = 'i2cset -y 10 0x24 0x24 0x22'
          os.system(i2cset_cmd)

          # Execute libcamera-still command
          libcamera_cmd = f'libcamera-still --immediate --width 1920 --height 1080 -e png  -o {dir}/ff{n}.png  --mode 3264:2448 --nopreview'
          os.system(libcamera_cmd)

          now = datetime.now().strftime('%H:%M:%S.%f')[:-3]
          sheet[f'B{n+1}'] = now
        
def function4():
          global n
          n += 1
          now = datetime.now().strftime('%H:%M:%S.%f')[:-3]
          sheet[f'A{n+1}'] = now
          i2cset_cmd = 'i2cset -y 10 0x24 0x24 0x32'
          os.system(i2cset_cmd)

          # Execute libcamera-still command
          libcamera_cmd = f'libcamera-still --immediate --width 1920 --height 1080 -e png  -o {dir}/ff{n}.png  --mode 3264:2448 --nopreview'
          os.system(libcamera_cmd)

          now = datetime.now().strftime('%H:%M:%S.%f')[:-3]
          sheet[f'B{n+1}'] = now

for i in range(5):
    function1()
    function2()
    function3()
    function4()
  

  
  
wb.save("sample_file205.xlsx")  

File: multi_cam_image_capture.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <chrono>
#include <iomanip>
#include <sstream>
#include <cstdlib>

using namespace std;

ofstream fs;
string filename = "exampleOutput100.csv";
const std::string dir = "/home/pi/indrone/test5/finalc1";

string time()
{
    // Get current time
    auto currentTime = std::chrono::system_clock::now();

    // Convert time to milliseconds since epoch
    auto ms = std::chrono::duration_cast<std::chrono::milliseconds>(currentTime.time_since_epoch());

    // Get the current time as a string with milliseconds
    std::time_t currentTimeSeconds = std::chrono::system_clock::to_time_t(currentTime);
    std::tm *timeInfo = std::localtime(&currentTimeSeconds);

    // Format the timestamp string
    std::stringstream timestamp;
    timestamp << std::put_time(timeInfo, "%H:%M.%S") << std::setfill('0') << std::setw(3) << ms.count() % 1000;

    // Return the timestamp as a string
    return timestamp.str();
}

int main()
{
    fs.open(filename);

    fs << "Start time"
       << ","
       << "end time" << std::endl;
    int a = 0;
    for (int i = 0; i < 50; i++)
    {
        if (a > 3)
        {
            a = 0;
        }
        string A = time();
        std::string i2cset_cmd = "i2cset -y 10 0x24 0x24 0x" + to_string(a)+ "2";
        std::system(i2cset_cmd.c_str());

        std::string command = "libcamera-still --immediate --width 1920 --height 1080 -e png -o " + dir + to_string(i) + ".png --mode 4056:3040";
        std::system(command.c_str());
        string B = time();

        fs << A << "," << B << std::endl;
        a++;
        
    }

    fs.close();
    return 0;
}
